tests/test_meta/test_grouped_transformer.py

```diff
@@ -0,0 +1,203 @@
+    X, y = random_xy_dataset_clf
+    X, y = check_X_y(X, y)
+
+    # This is not a transformer
+    trf = LinearRegression()
+    transformer = GroupedTransformer(trf)
+
+    with pytest.raises(ValueError):
+        transformer.fit(X, y)
+
+
+def test_multiple_grouping_columns(random_xy_dataset_clf):
+    X, y = random_xy_dataset_clf
+    X, y = check_X_y(X, y)
+
+    # Make sure all groups are present
+    groups = np.tile(
+        # 4x2 array, repeated until it fits
+        np.array([[0, 0], [0, 1], [1, 0], [1, 1]]),
+        reps=((X.shape[0] + 3) // 4, 1)
+    )[:X.shape[0], :]
+    X_with_groups = np.concatenate([groups, X], axis=1)
+
+    # Some weird interval to make sure we test the right values
+    scaling_range = (13, 42)
+
+    trf = MinMaxScaler(scaling_range)
+    transformer = GroupedTransformer(clone(trf), groups=(0, 1))
+    transformed = transformer.fit(X_with_groups, y).transform(X_with_groups)
+
+    assert transformed.shape == X.shape
+
+    df_with_groups = pd.concat([
+        pd.DataFrame(groups, columns=["A", "B"]),
+        pd.DataFrame(transformed)
+    ], axis=1)
+
+    assert np.allclose(df_with_groups.groupby(["A", "B"]).min(), scaling_range[0])
+
+    # If a group has a single element, it defaults to min, so check wether all maxes are one of the bounds
+    maxes = df_with_groups.groupby(["A", "B"]).max()
+    assert np.all(
+        np.isclose(maxes, scaling_range[1]) | np.isclose(maxes, scaling_range[0])
+        # We have at least some groups larger than 1, so there we should find the max of the range
+    ) and np.any(np.isclose(maxes, scaling_range[1]))
+
+
+def test_missing_groups_transform_global(random_xy_dataset_clf):
+    X, y = random_xy_dataset_clf
+    X, y = check_X_y(X, y)
+
+    # Make sure all groups are present
+    groups = np.repeat([0, 1], repeats=(X.shape[0] + 1) // 2)[:X.shape[0], np.newaxis]
+    X_with_groups = np.concatenate([groups, X], axis=1)
+
+    # Some weird interval to make sure we test the right values
+    scaling_range = (13, 42)
+
+    trf = MinMaxScaler(scaling_range)
+    transformer = GroupedTransformer(clone(trf), groups=0)
+    transformer.fit(X_with_groups, y)
+
+    # Array with 2 rows, first column a new group. Remaining top are out of range so should be the range
+    X_test = np.concatenate([
+        np.array([[3], [3]]), np.stack([X.min(axis=0), X.max(axis=0)], axis=0)
+    ], axis=1)
+
+    transformed = transformer.transform(X_test)
+
+    # Top row should all be equal to the small value of the range, bottom the other
+    assert np.allclose(transformed[0, :], scaling_range[0])
+    assert np.allclose(transformed[1, :], scaling_range[1])
+
+
+def test_missing_groups_transform_noglobal(random_xy_dataset_clf):
+    X, y = random_xy_dataset_clf
+    X, y = check_X_y(X, y)
+
+    # Make sure all groups are present
+    groups = np.repeat([0, 1], repeats=(X.shape[0] + 1) // 2)[:X.shape[0], np.newaxis]
+    X_with_groups = np.concatenate([groups, X], axis=1)
+
```

Comment:
You're currently using the `random_xy_dataset_clf` which is fine, but I don't think you really need it for the tests here. I also think that you can cover the same ground by introducing a fixture in this file that generates `X, y, groups, X_with_groups`. That way we won't need this code block everywhere:

```
    X, y = random_xy_dataset_clf
    X, y = check_X_y(X, y)

    # Make sure all groups are present
    groups = np.repeat([0, 1], repeats=(X.shape[0] + 1) // 2)[:X.shape[0], np.newaxis]
    X_with_groups = np.concatenate([groups, X], axis=1)
```


Would result in less code while still testing the same material. 

It should also massively speed up the tests.

Reply:
Good point, I agree. Will change it.